main/src/library/Fixpoint/Compiler.flix

```diff
@@ -31,16 +31,15 @@ namespace Fixpoint {
     /// This means to compile all rules in stratum 0, then stratum 1 and so on.
     /// A rule belongs to the same stratum as its head predicate.
     ///
-    def compile(cs: Datalog[v], strat: Map[PredSym, Int32]): RamStmt[v] with ToString[v] = match cs {
-        case Datalog(facts, rules) =>
-            let evalFacts = Array.fold(Array.map(compileRule, facts)) as & Pure;
-            let evalStrata =
-                Map.invert(strat) |>
-                Map.map(restrict(rules) >> compileStratum) |>
-                Map.valuesOf |>
-                List.fold;
-            List.fold(evalFacts :: evalStrata :: Nil)
-    }
+    def compile(cs: Datalog[v], strat: Map[PredSym, Int32]): RamStmt[v] with ToString[v] =
+        let doIt = match Datalog(facts, rules) -> {
+            let stmts = MutList.new();
+            Array.foreach(compileRule(stmts), facts);
+            Map.invert(strat) |>
+            Map.foreach(_ -> restrict(rules) >> compileStratum(stmts));
```

Comment:
I guess I am confused why the argument passed to foreach is not used?

Reply:
The first argument is the key. The map is from strata to predicates, i.e. `Map[Int32, PredSym]`. We don't care what number a stratum has, only that we compile them in order.

Reply:
OK, a future comment would be good, but ill merge now.